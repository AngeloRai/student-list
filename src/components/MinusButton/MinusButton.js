import { ImMinus } from "react-icons/im";

const MinusButton = ({ setIsCollapsed }) => {
  return (
    <button
      className="-mt-6 text-2xl text-slate-500 hover:text-black"
      onClick={() => setIsCollapsed(false)}
    >
      <ImMinus />
    </button>
  );
};

export default MinusButton;
