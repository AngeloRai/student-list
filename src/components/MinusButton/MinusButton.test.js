import React from "react";
import { shallow, mount } from "enzyme";
import MinusButton from "./MinusButton";
import toJson from "enzyme-to-json";

describe("tests button component", () => {
    it("renders without crashing", () => {
    shallow(<MinusButton />);
  });

  it("checks for changes in the code", () => {
    const wrapper = shallow(<MinusButton />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("accepts props", () => {
    const mockSetIsCollapsed = jest.fn();
    const wrapper = mount(<MinusButton setIsCollapsed={mockSetIsCollapsed} />);
    expect(wrapper.props().setIsCollapsed).toEqual(mockSetIsCollapsed);
  });

  it("tests if function invokes when it's clicked", async () => {
    const mockSetIsCollapsed = jest.fn()
    const wrapper = mount(<MinusButton setIsCollapsed={mockSetIsCollapsed} />);
    wrapper.find("button").simulate("click");
    expect(mockSetIsCollapsed).toHaveBeenCalled();
  });
  
});
