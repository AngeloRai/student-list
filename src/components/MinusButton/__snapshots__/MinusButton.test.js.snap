// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests button component checks for changes in the code 1`] = `
<button
  className="-mt-6 text-2xl text-slate-500 hover:text-black"
  onClick={[Function]}
>
  <ImMinus />
</button>
`;
